Tests/AsyncHTTPClientTests/Mocks/MockConnectionPool.swift

```diff
@@ -0,0 +1,486 @@
+
+        self.connections[connection.id] = nil
+    }
+
+    mutating func startConnectionBackoffTimer(_ connectionID: Connection.ID) throws {
+        guard self.connections[connectionID] == nil else {
+            throw Errors.connectionExists
+        }
+
+        guard !self.backoff.contains(connectionID) else {
+            throw Errors.connectionBackoffTimerExists
+        }
+
+        self.backoff.insert(connectionID)
+    }
+
+    mutating func connectionBackoffTimerDone(_ connectionID: Connection.ID) throws {
+        guard self.backoff.remove(connectionID) != nil else {
+            throw Errors.connectionBackoffTimerNotFound
+        }
+    }
+
+    mutating func cancelConnectionBackoffTimer(_ connectionID: Connection.ID) throws {
+        guard self.backoff.remove(connectionID) != nil else {
+            throw Errors.connectionBackoffTimerNotFound
+        }
+    }
+
+    // MARK: Connection destruction
+
+    /// Closing a connection signals intend. For this reason, it is verified, that the connection is not running any
+    /// requests when closing.
+    mutating func closeConnection(_ connection: Connection) throws {
+        guard var mockConnection = self.connections.removeValue(forKey: connection.id) else {
+            throw Errors.connectionNotFound
+        }
+
+        try mockConnection.close()
+    }
+
+    /// Aborting a connection does not verify if the connection does anything right now
+    mutating func abortConnection(_ connectionID: Connection.ID) throws {
+        guard self.connections.removeValue(forKey: connectionID) != nil else {
+            throw Errors.connectionNotFound
+        }
+    }
+
+    // MARK: Connection usage
+
+    mutating func parkConnection(_ connectionID: Connection.ID) throws {
+        guard var connection = self.connections[connectionID] else {
+            throw Errors.connectionNotFound
+        }
+
+        try connection.park()
+        self.connections[connectionID] = connection
+    }
+
+    mutating func activateConnection(_ connectionID: Connection.ID) throws {
+        guard var connection = self.connections[connectionID] else {
+            throw Errors.connectionNotFound
+        }
+
+        try connection.activate()
+        self.connections[connectionID] = connection
+    }
+
+    mutating func execute(_ request: HTTPSchedulableRequest, on connection: Connection) throws {
+        guard var connection = self.connections[connection.id] else {
+            throw Errors.connectionNotFound
+        }
+
+        try connection.execute(request)
+        self.connections[connection.id] = connection
+    }
+
+    mutating func finishExecution(_ connectionID: Connection.ID) throws {
+        guard var connection = self.connections[connectionID] else {
+            throw Errors.connectionNotFound
+        }
+
+        try connection.finishRequest()
+        self.connections[connectionID] = connection
+    }
+
+    mutating func randomStartingConnection() -> Connection.ID? {
```

Comment:
These are all for the fuzzing, presumably? Might be worth sticking them in an extension.